stock_picking_group_by_partner_by_carrier/models/stock_move.py

```diff
@@ -100,6 +101,11 @@ def _domain_search_picking_handle_move_type(self):
         return [("move_type", "=", self.group_id.move_type)]
 
     def _key_assign_picking(self):
+        if (
+            not self.picking_id.picking_type_id.group_pickings
+            and self.picking_id.partner_id.disable_picking_grouping
```

Comment:
the previous diff chunk as an "or" and I think it is correct. Can you check?
```suggestion
            or self.picking_id.partner_id.disable_picking_grouping
```

Reply:
suggestion: have a helper method _is_grouping_disabled returning 

```
    not self.picking_type_id.group_pickings
    or self.partner_id.disable_picking_grouping
```

Reply:
Agreed

Reply:
:+1: